### PR TITLE
Misc XAML generator cleanup

### PR DESCRIPTION
**Pull request by Youssef1313**
GitHub Issue (If applicable): closes #

## PR Type

What kind of change does this PR introduce?

### <samp>🤖 Generated by Copilot at 6101942</samp>

This pull request cleans up and optimizes the code of the XAML source generators in the `Uno.UI.SourceGenerators` project. It removes unused and redundant code, improves performance and formatting, and adds a new method to the `IndentedStringBuilder` class. It also simplifies the telemetry event for the XAML code generation.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [ ] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/XamlCodeGeneration.Telemetry.cs [hunk 1]
```diff
@@ -107,7 +107,6 @@ private void TrackStartGeneration(XamlFileDefinition[] files)
 					_telemetry.TrackEvent(
 						"generate-xaml",
 						new[] {
-							("IsWasm", _isWasm.ToString()),
```
Comment (Youssef1313) [inline src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/XamlCodeGeneration.Telemetry.cs:110]: Note: This piece of info will be available in UnoRuntime value. So we haven't actually lost that info.

File: src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/XamlCodeGeneration.cs [hunk 2]
```diff
@@ -687,16 +684,6 @@ private SourceText GenerateGlobalResources(IEnumerable<XamlFileDefinition> files
...
 
-			// If a failure happens here, this means that the _isWasm was not properly set as the DefineConstants msbuild property
-			// was not populated. This can happen when the property is set through a target with the "CreateProperty" task, and the
-			// Uno.SourceGeneration tasks do not execute this task properly.
-			if (!_isWasm)
-			{
-				writer.AppendLineIndented("#if __WASM__");
-				writer.AppendLineIndented("#error invalid internal source generator state. The __WASM__ DefineConstant was not propagated properly.");
-				writer.AppendLineIndented("#endif");
-			}
```
Comment (Youssef1313) [inline src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/XamlCodeGeneration.cs:698]: This is not going to happen, ever. It's just an unnecessary generated code. From the comment above, this used to be an issue specific to Uno.SourceGeneration which is no longer used.

Now, the compiler will always pass us a correct value for DefineConstants.

File: src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/XamlCodeGenerator.cs [hunk 3]
```diff
@@ -29,25 +20,17 @@ public void Execute(GeneratorExecutionContext context)
 			//	Debugger.Launch();
 			//}
 
-			//
-			// Lock the current generator instance, as it may be invoked concurrently
-			// in the context of omnisharp when saving and editing fast enough, causing
-			// corruption issues in the GenerationRunInfoManager.
```
Comment (Youssef1313) [inline src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/XamlCodeGenerator.cs:35]: The locking was added for something specific to GenerationRunInfoManager which has been removed.

File: src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/XamlCodeGeneration.Telemetry.cs (hunk 1 above)
Comment (jeromelaban) [reply src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/XamlCodeGeneration.Telemetry.cs:110]: Let's keep it for now, even if duplicated.

**Review by jeromelaban — APPROVED** [review]

**Pull request merged**